queries/scala/highlights.scm

```diff
@@ -135,18 +154,29 @@
   "val"
   "var"
   "with"
+  "given"
+  "end"
+  "implicit"
+  "extension"
+  "with"
 ] @keyword
 
 [
   "abstract"
   "final"
-  "implicit"
+  "using"
   "lazy"
-  "private"
-  "protected"
   "sealed"
 ] @type.qualifier
 
+(inline_modifier) @storageclass
+
+[
+  "private"
+  "protected"
+] @storageclass
```

Comment:
Access modifiers shouldn't be `@storageclass` unless they affect storage in memory.

Reply:
Is there sort of a definitive guide stating this somewhere? From the contributing docs it says:

> @storageclass          ; visibility/life-time modifiers

And when this was added in https://github.com/tree-sitter/tree-sitter-scala/pull/138 we also saw similar verbiage in the sublime docs.

Reply:
Alrighty, I've reverted these back.

Reply:
Note that in the TextMate intro blog post https://macromates.com/blog/2005/introduction-to-scopes/ Allan Odgaard wrote

> The tags are implicit in TextMate, using the language grammar. So if we apply this example grammar:

```
scopeName = 'c_source';
patterns = (
    { name = 'storage'; match = 'char'; },
    { name = 'modifier'; match = 'const'; },
    { name = 'operator'; match = '[=*]'; },
    { name = 'string'; begin = '"'; end = '"';
        patterns = (
            { name = 'constant'; match = '\\\\.'; },
        );
    }
);
```

> To this line of code:

```c
char const* str = "Hello world\n";
```

So in TextMate terminology, "storage" refers not exactly just to memory storage, but what in statically-typed programming language community would call "type" if not "field". [Sublime](https://www.sublimetext.com/docs/scope_naming.html#storage) therefore uses `storage.modifier` to describe things like `static`, `inline`, `const`, `public` and `private`. They are modifiers that can be applied to fields.

Do we have a similar concept here?

Reply:
> Do we have a similar concept here?

Not yet.